### PR TITLE
Reduce hooks in useValue

### PR DESCRIPTION
**Pull request by steveruizok**
This PR simplifies `useValue` (in the same spirit of `useReactor`). Previously, we were using two memos and a ref; now it uses one memo.

### Change type

- [x] `improvement`

### Release notes

- Small improvement to the performance on `useValue`.

### CODE REVIEW

File: packages/state-react/src/lib/useValue.ts [hunk 1]
```diff
@@ -49,38 +49,36 @@ export function useValue() {
 	const deps = args.length === 3 ? args[2] : [args[0]]
 	const name = args.length === 3 ? args[0] : `useValue(${args[0].name})`
 
-	const isInRender = useRef(true)
-	isInRender.current = true
+	let isInRender = true
```
Comment (ds300) [inline packages/state-react/src/lib/useValue.ts:52]: this is not equivalent to using the useRef, and it won't work as intended because this let binding is scoped to each render individually while a useRef is scoped to the component's lifetime. setting isInRender to true will have no effect on a computed that was created during a previous render.

**Review by ds300 — CHANGES_REQUESTED** [review]

File: packages/state-react/src/lib/useValue.ts (hunk 1 above)
Comment (steveruizok) [reply packages/state-react/src/lib/useValue.ts:52]: I might be getting tripped up on react lifecycle here. I can drop this back to use the ref.

You've got a unit test for this edge case though, right? It passes on this PR, which is weird if the ref is needed.

Comment (ds300) [reply packages/state-react/src/lib/useValue.ts:52]: your change here actually affects untested behvaior. namely that the useValue should throw synchronously during render if the computed throws an error and the component is still alive.

And now that I try to test this I realise it's not working as expected. lemme see if i can come up with a fix and I'll make an alternative pr to this

**Comment by steveruizok** [comment]
Much better in #6405 🙌
